recipe/0000-unvendor_antlr.patch

```diff
@@ -0,0 +1,20 @@
+--- pypi-upstream/build_helpers/build_helpers.py	2024-09-06 16:54:09.640879247 +0200
++++ pypi-upstream/build_helpers/build_helpers.py	2024-09-06 16:26:54.958978440 +0200
+@@ -176,7 +176,7 @@
+ 
+     def run(self) -> None:
+         """Run command."""
+-        root_dir = abspath(dirname(__file__))
++        root_dir = os.environ['PREFIX']
+         project_root = abspath(dirname(basename(__file__)))
+         for grammar in [
+             "hydra/grammar/OverrideLexer.g4",
+@@ -185,7 +185,7 @@
+             command = [
+                 "java",
+                 "-jar",
+-                join(root_dir, "bin/antlr-4.9.3-complete.jar"),
++                join(root_dir, "lib/antlr4-4.9.3-complete.jar"),
```

Comment:
This seems to be breaking the build:

```
raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['java', '-jar', '/opt/conda/conda-bld/hydra-core_1725954885436/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/antlr4-1.3.2-complete.jar', '-Dlanguage=Python3', '-o', '/opt/conda/conda-bld/hydra-core_1725954885436/work/pypi-upstream/hydra/grammar/gen/', '-Xexact-output-dir', '-visitor', '/opt/conda/conda-bld/hydra-core_1725954885436/work/pypi-upstream/hydra/grammar/OverrideLexer.g4']' returned non-zero exit status 1.
  error: subprocess-exited-with-error
```


```
ERROR: Failed building wheel for hydra-core
Failed to build hydra-core
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (hydra-core)
```

Reply:
I believe we were running into a similar issue which is why we hardcore the version in.

Reply:
`os.environ['PKG_VERSION']` did not work because it's the actual package version (`1.3.2`) and not the version of `antlr` that we want to use. I pushed a change that uses an ad-hoc environment variable `ANTLR_VERSION` set in the recipe. The version of `antlr` and `antlr4-python3-runtime` should be in sync anyway.

Reply:
Awesome! Thanks @lorepirri!